src/Microsoft.AspNet.Razor/TagHelpers/TagHelperDescriptorProvider.cs

```diff
@@ -58,23 +59,41 @@ public IEnumerable<TagHelperDescriptor> GetTagHelpers(string tagName)
                 descriptors = new HashSet<TagHelperDescriptor>(TagHelperDescriptorComparer.Default);
             }
 
-            // If the requested tag name is the catch-all target, we should short circuit.
-            if (tagName.Equals(CatchAllDescriptorTarget, StringComparison.OrdinalIgnoreCase))
+            // If the requested tag name is the catch-all target, we shouldn't do the work of concatenating extra
+            // descriptors.
+            if (!tagName.Equals(CatchAllDescriptorTarget, StringComparison.OrdinalIgnoreCase))
```

Comment:
having trouble believing this check (or the one it replaces) is necessary.  is there a realistic scenario in which the `tagName` will be anything other than a `string` from the Razor source?